src/util/firebase/chain.ts

```diff
@@ -2,6 +2,18 @@ import db from "./firebaseConfig";
 
 import { IChain } from "../../types";
 
+const getChain = async (chainId: string) => {
+  return await db.collection("chains")
+    .doc(chainId)
+    .get()
+    .then((doc) => {
+      return {
+        chain: doc.data() as IChain,
+        chainReference: doc.ref
+      };
+    });
+};
```

Comment:
You're mixing two different types of async processing here. The .get() call should already return a promise, you can you await that call. "then" uses the callback-style of async processing. You can mix them.. but it doesn't really make things clearer.

Some examples here: https://firebase.google.com/docs/firestore/query-data/get-data#node.js_6 -> click on javascript tabs.

Reply:
Thanks for the explanation and docs! I've tried to rewrite it, please take a look.

Reply:
Looks good now.